<?xml version="1.0"?>

<launch>

  <!-- arguments -->
  <arg name="uri" default="radio://0/84/1M/E7E7E7E702" />
  <arg name="world_frame" default="/world" />
  <arg name="cf_frame" default="cf2" />
  <arg name="prefix" default="cf2"/>
  <arg name="server" default="198." />
  <arg name="port" default="3883" />

  <!-- start crazyflie_client -->
  <include file="$(find crazyflie_driver)/launch/crazyflie_server.launch">
  </include>

  <!-- the first crazyflie -->
  <group ns="crazyflie">

    <!-- add crazyflie -->
    <include file="$(find crazyflie_driver)/launch/crazyflie_add.launch">
      <arg name="uri" value="$(arg uri)" />
      <arg name="tf_prefix" value="$(arg prefix)" />
      <arg name="enable_logging" value="True" />
    </include>

    <!-- add inner-loop controller -->
    <include file="$(find crazyflie_controller)/launch/crazyflie2.launch">
      <arg name="frame" value="$(arg cf_frame)" />
    </include>

    <!-- add outer-loop controller, hover only for now -->
    <node name="cf2DSI_controller" pkg="crazyflie_2DSI" type="controller.py" output="screen">
      <param name="cmd_topic" value="$(arg prefix)"+"/cmd_op" />
      <param name="goal_topic" value="$(arg prefix)"+"/goal" />
    </node>

    <!-- run vrpn client -->
<!--     <include file="$(find vrpn_client_ros)/launch/sample.launch">
      <arg name="server" value="$(arg server)" />
    </include> -->
    <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
      <rosparam subst_value="true">
        server: $(arg server)
        port: $(arg port)
        update_frequency: 100.0
        frame_id: $(world_frame)
        child_frame_id: $(arg cf_frame)
        use_server_time: false
        broadcast_tf: True
        refresh_tracker_frequency: 1.0
      </rosparam>
    </node>
  </group>

</launch>